src/lib/network_pool/network_pool_base.ml

```diff
@@ -130,22 +131,27 @@ end)
             ] ;
         drop diff' rejected cb )
       else (
-        [%log' debug t.logger] "Rebroadcasting diff %s"
-          (Resource_pool.Diff.summary diff') ;
+        [%log debug] "Rebroadcasting diff %s" (Resource_pool.Diff.summary diff') ;
         forward t.write_broadcasts diff' rejected cb )
     in
     O1trace.sync_thread apply_and_broadcast_thread_label (fun () ->
         match Resource_pool.Diff.unsafe_apply t.resource_pool diff with
         | Ok (`Accept, accepted, rejected) ->
+            Resource_pool.Diff.log_internal ~logger "accepted" env ;
             rebroadcast (accepted, rejected)
         | Ok (`Reject, accepted, rejected) ->
+            Resource_pool.Diff.log_internal ~logger "rejected"
+              ~reason:"not_applied" env ;
             Broadcast_callback.reject accepted rejected cb
         | Error (`Locally_generated res) ->
+            Resource_pool.Diff.log_internal ~logger "rejected"
+              ~reason:"locally_generated" env ;
             rebroadcast res
         | Error (`Other e) ->
             [%log' debug t.logger]
               "Refusing to rebroadcast. Pool diff apply feedback: $error"
               ~metadata:[ ("error", Error_json.error_to_yojson e) ] ;
+            Resource_pool.Diff.log_internal ~logger "rejected" env ;
```

Comment:
I tried to make "reason" a very short tag, so that it's easy to query/group, whereas `e` can be converted only to a long string that is not well formatted.